tests/elections/test_election.py

```diff
@@ -0,0 +1,64 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from bigchaindb.elections.election import Election
+
+
+@pytest.mark.bdb
+def test_approved_elections_one_migration_one_upsert(
+        b,
+        ongoing_validator_election, validator_election_votes,
+        ongoing_chain_migration_election, chain_migration_election_votes
+):
+    txns = validator_election_votes + \
+           chain_migration_election_votes
+    mock_chain_migration, mock_store_validator = run_approved_elections(b, txns)
+    mock_chain_migration.assert_called_once()
+    mock_store_validator.assert_called_once()
+
+
+@pytest.mark.bdb
+def test_approved_elections_one_migration_two_upsert(
+        b,
+        ongoing_validator_election, validator_election_votes,
+        ongoing_validator_election_2, validator_election_votes_2,
+        ongoing_chain_migration_election, chain_migration_election_votes
+):
+    txns = validator_election_votes + \
+           validator_election_votes_2 + \
+           chain_migration_election_votes
+    mock_chain_migration, mock_store_validator = run_approved_elections(b, txns)
+    mock_chain_migration.assert_called_once()
+    mock_store_validator.assert_called_once()
```

Comment:
This test should be done without mocking i.e. the test should assert that,
- First validator election concludes and updates the validator collection
- Second validator election doesn't update the validator collection
- The chain migration is concluded and the new chain is read and assert form `abci_chains` collection